packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -43,16 +46,41 @@ library LibProposing {
             msg.sender != resolver.resolve("solo_proposer", false)
         ) revert L1_NOT_SOLO_PROPOSER();
 
-        if (txList.length > config.maxBytesPerTxList) revert L1_TX_LIST();
+        if (input.txListEnd <= input.txListStart) revert L1_TX_LIST_RANGE();
 
         if (
             input.beneficiary == address(0) ||
             input.gasLimit > config.blockMaxGasLimit
         ) revert L1_INVALID_METADATA();
 
-        // We need txListHash as with EIP-4844, txList is no longer
-        // accssible to EVM.
-        if (input.txListHash != keccak256(txList)) revert L1_TX_LIST_HASH();
+        uint64 _now = uint64(block.timestamp);
+        uint32 _size = uint32(txList.length);
+
+        if (_size == 0) {
```

Comment:
Not sure I understand what this is trying to do. Is this trying to emulate data blobs using tx data?

Reply:
Yes, that's the goal. Do you think it make sense?

Reply:
From a practical point of view, the decentralized sequencer aspect complicates this a bit I think because currently you kind of only want to commit data on-chain when you know when you are also the one that actually uses in a block, otherwise a proposer simply paid the data fee for another proposer to use in the future. It's currently as efficient, or even more efficient because more flexible, to just push the data for each block separately so no real reason to use this system.

But if we just want to test the system as close as possible to EIP-4844 because that's what eventually will be used it does make sense to me to do this. But then I would decouple submitting the tx data from the block proposal completely to more closely resemble blob transactions.

Reply:
I guess I don't know yet whether with EIP-4844, the blob must be submitted using a separate transaction or any transaction can just attach a blob with it. Do you know?

Reply:
Oh I assumed it would be completely separate because the blob transactions are a separate transaction type and have their own mempool, but they seem to still have a `to` address and I don't know what the purpose of that is. So not sure if blob transactions can execute smart contract code or not.

Reply:
Okay looks like it's basically a normal transaction with blob data attached, I got this completely wrong. It is possible there could be some small limitations: https://ethereum-magicians.org/t/eip-4844-shard-blob-transactions/8430/34.